File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.cc
#include <bits/stdc++.h>
using namespace std;

int arr[30], flag;
string s, start, mid, last;

int main() {
  cin >> s;

  for (int i = 0; i < s.size(); i++) {
    arr[s[i] - 'A']++;
  }

  for (int i = 0; i < 30; i++) {
    for (int j = 0; j < arr[i] - 1; j += 2) {
      start += i + 'A';
      last = char(i + 'A') + last;
    }

    if (arr[i] & 1) {
      flag++;
      if (flag > 1) {
        cout << "I'm Sorry Hansoo";
        return 0;
      }
      mid += i + 'A';
    }
  }

  cout << start + mid << last;

  return 0;
}

File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.js
const input = require('fs').readFileSync(process.platform === "linux" ? '/dev/stdin' : './input.txt').toString().trim().split('');

const cnt = Array(30).fill(0);

function answer (input) {
    for(const c of input){
        cnt[c.charCodeAt(0) - 65]++;
    }
    let start = "", mid= "", last = "";
    let flag = 0;

    for (let i=0; i<30; i++){
        for(let j=0; j<cnt[i] - 1; j+=2){
            start += String.fromCharCode(65+i);
            last = String.fromCharCode(65+i) + last;
        }

        if(cnt[i] & 1){
            flag++;
            if(flag > 1){
                console.log("I'm Sorry Hansoo");
                return;
            }
            mid += String.fromCharCode(65+i);
        }
    }
    console.log(start+ mid + last);
}
answer(input);